src/app/journey/middleware.js

```diff
@@ -21,8 +21,18 @@ async function journeyApi(action, ipvSessionId) {
 
 async function handleJourneyResponse(action, res, ipvSessionId) {
   const response = await journeyApi(action, ipvSessionId);
-  if (response.data?.redirect?.event) {
-    await handleJourneyResponse(response?.data?.redirect?.event, res);
+  if (response.data?.redirect) {
+    if(response.data?.redirect?.event) {
```

Comment:
The returned data structure won't be in this format. I think it will just things like:

```suggestion
    if(response.data?.journey) {
```

```suggestion
    if(response.data?.page) {
```

```suggestion
    if(response.data?.cri) {
```

Reply:
Will be resolved in a separate ticket as we were working with the main branch version of the openApi